### ISSUE TITLE
`resolveId` expanding mapped network drives in file names

### ISSUE THREAD

**Issue by stephenlrandall** [sveltejs/vite-plugin-svelte#525]
### Describe the bug

There is a discrepancy in resolved path names between `vite:resolve` and `vite-plugin-svelte`. The latter's plugin function `resolveId` is using `findDepPkgJsonPath` from "vitefu", which expands mapped drive names. Looking at a particular dependency:

```
dep: svelte-select
parent: Z:/<repo path>/src/components/elements/Select.svelte?id=0
dep data path: \\<network drive>\<repo path>\node_modules\svelte-select\package.json
```

This is returned to Vite as the `resolved` object:

```
{ id: '/<network drive>/<repo path>/node_modules/svelte-select/index.js' }
```

By contrast, a non-`package.svelte` dependency (for example, `math.js`) is skipped and handled by `vite:resolve` instead:

```
{ id: 'Z:/<repo path>/node_modules/mathjs/lib/esm/index.js' }
```

This discrepancy leads to the "file not found" error shown in the log section below.

---

I wasn't sure if this issue would better fit here or in [vitefu](https://github.com/svitejs/vitefu). But [this comment](https://github.com/sveltejs/vite-plugin-svelte/issues/501#issuecomment-1318869724) referenced using Vite's `resolve`, which -- if that's the same as `vite:resolve` -- would fix this issue.

### Reproduction URL

not reproducible with a repo alone

### Reproduction

On a mapped network drive, set up a new SvelteKit project with at least one dependency that uses the `svelte` package field.

### Logs

```shell
node:fs:591
  handleErrorFromBinding(ctx);
  ^

Error: ENOENT: no such file or directory, open '/<network drive>/<repo path>/node_modules/svelte-select/index.js'
    at Object.openSync (node:fs:591:3)
    at Object.readFileSync (node:fs:459:35)
    at extractExportsData (file:///Z:/<repo path>/node_modules/vite/dist/node/chunks/dep-5605cfa4.js:42115:31)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  errno: -4058,
  syscall: 'open',
  code: 'ENOENT',
  path: '/<network drive>/<repo path>/node_modules/svelte-select/index.js'
}
```

### System Info

```shell
System:
    OS: Windows 10
    CPU: (6) x64 Intel(R) Xeon(R) Gold 6226R CPU @ 2.90GHz
    Memory: 3.91 GB / 12.00 GB
  Binaries:
    Node: 18.2.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.9.0 - C:\Program Files\nodejs\npm.CMD
  npmPackages:
    @sveltejs/adapter-static: ^1.0.0-next.48 => 1.0.0-next.48
    @sveltejs/kit: 1.0.0-next.572 => 1.0.0-next.572
    svelte: ^3.53.1 => 3.53.1
    vite: 3.2.5 => 3.2.5
```

**Comment by bluwy** [comment]
@stephenlrandall if you can send a PR to vitefu with a fix, that'll be great! it would likely need a fix [here](https://github.com/svitejs/vitefu/blob/8b04a82839eed0d36459ba3443d8bd32f3700bf2/src/index.js#L196-L217). I'm not sure how a leading backslash happens but if the fix can be done for Windows specifically it should be good.

**Comment by stephenlrandall** [comment]
@bluwy Found something I didn't expect. If you add some print statements to the function you linked:

```js
console.log(`pkg: ${pkg}`);
try {
  await fs.access(pkg);

  // -------------------------------------------------
  console.log(`sync: ${fsSync.realpathSync(pkg)}`);
  const asyncRes = await fs.realpath(pkg);
  console.log(`async: ${asyncRes}`);
  // -------------------------------------------------

  return await fs.realpath(pkg);
}
```

what you get is:

```
pkg: Z:\<repo path>\node_modules\svelte-select\package.json
sync: Z:\<repo path>\node_modules\svelte-select\package.json
async: \\<network drive>\<repo path>\node_modules\svelte-select\package.json
```

Vite, in `getRealPath` within `vite:resolve`,  uses `realpathSync` from `node:fs`. `vitefu` uses `realpath` from `node:fs/promises`. As to why these functions have different behavior, there's some discussion in [this issue](https://github.com/nodejs/node/issues/37737). (The relevant functions are [here](https://github.com/nodejs/node/blob/main/lib/fs.js), for reference.)

As a fix (and to at least be consistent with Vite proper), would it be possible to simply replace `node:fs/promises` with the synchronous `node:fs`? If so, I can open a PR in `vitefu` with that change.

**Comment by bluwy** [comment]
Published vitefu v0.2.4 with the fix. When freshly installing vite-plugin-svelte, it should get the latest version of it now. Closing as fixed. Thanks again for debugging this!
